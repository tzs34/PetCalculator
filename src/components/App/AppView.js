import {Component} from 'react'
import styled from 'styled-components'
import { FlexRow } from 'utils/styles'
import Header from '../Display/Header'
import ButtonBar from '../Display/ButtonBar'
import BalanceAndLabel from '../Display/HeaderLabel'
import PetView from '../Display/PetView'
import LoadingHOC from '../Hoc/LoadingIndicatorHOC'

const Container = styled.div`
    position: relative;
    display: flex;
    flex-direction: column;
    width: 100%;
    height: 100%;
  `

const ButtonBarContainer = styled.div`
  margin-top: 80px;
`
@LoadingHOC
class AppView extends Component{

  render(){
    return (
      <Container>
        <BalanceAndLabel />
        <ButtonBarContainer>
          <PetView />
        </ButtonBarContainer>
      </Container>
    )
  }
}

export default AppView
