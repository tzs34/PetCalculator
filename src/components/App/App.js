import {Component, createContext} from 'react'
import PropTypes from "prop-types"
import AppView from './AppView'
import styled, { injectGlobal, ThemeProvider } from 'styled-components'
import Copy from 'utils/copy'
import { FlexRow} from 'utils/styles'
import AppContext from 'context/AppContext'
import {getPets} from '../../services/service'

injectGlobal`

  @import url('https://fonts.googleapis.com/css?family=Fredoka+One');

  body {
      margin: 0;
      font-family: 'Fredoka One', cursive;
      font-size: 2em;
      color: white;
    }
`
const Container = styled.div`
  width: 100%;
  height: 100%;
`

class App extends Component{

  state = {
    authenticated: true,
    data: [],
    pet: null
  }

  componentDidMount(){

    let pets = getPets()
    .then(({data}) => {
      this.setState({data})
    })
    .catch((e) => {
        // log error
    })

  }

  choosePet = ({target:{id}}) => {

    if(id !== this.state.pet){
      this.setState({pet: id})
    }

  }

  render(){

    let { authenticated, pet, data} = this.state
    let {choosePet} = this
    let petData = pet ? data.filter( animal => animal.id === pet)[0]: null

    return(
       <AppContext.Provider value={{authenticated, choosePet, petData}}>
         <Container>
           <AppView authenticated={authenticated}/>
         </Container>
      </AppContext.Provider>
    )
  }
}

export default App
