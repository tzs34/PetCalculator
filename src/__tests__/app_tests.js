import App from "../components/App/App";

test("App should render correctly", () => {
  expect(<App />).toMatchSnapshot();
});
