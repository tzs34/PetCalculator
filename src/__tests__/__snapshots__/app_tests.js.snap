// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div>
  <div
    class="App__Container-sc-1aluvzh-0 dydTnG"
  >
    <div>
      <div
        class="LoadingIndicatorHOC__OuterLoaderContainer-sc-3f7q3l-0 kcSYgy"
      >
        <div
          class="LoadingIndicatorHOC__LoaderContainer-sc-3f7q3l-1 iZHV"
        >
          <span
            class="isvg loading"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
